### PR TITLE
feat: update webpack-srouces types

### PR DESCRIPTION
**Pull request by SyMind**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: (https://github.com/webpack/webpack/blob/400db90db807d630b6c9f5e7ccde2b8f01690f3b/types.d.ts#L13331)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/webpack-sources/lib/CompatSource.d.ts
1. types/webpack-sources/lib/ConcatSource.d.ts
1. types/webpack-sources/lib/OriginalSource.d.ts
1. types/webpack-sources/lib/PrefixSource.d.ts
1. types/webpack-sources/lib/SourceMapSource.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e48d99e51be5e15b8367dc518858916ce6b7e79a
</p>

**Comment by peterblazejewicz** [comment]
hey, please include reference to original source change (docs?) and, if needed, bump a version

File: types/webpack-sources/lib/CompatSource.d.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import Source = require("./Source");
 
 declare class CompatSource extends Source {
     constructor(sourceLike: SourceLike);
-    static from(sourceLike: SourceLike): Source | CompatSource;
+    static from(sourceLike: SourceLike): Source;
```
Comment (gabritto) [inline types/webpack-sources/lib/CompatSource.d.ts:6]: Documentation says the previous type is correct: https://github.com/webpack/webpack-sources?tab=readme-ov-file#public-methods-4

**Review by gabritto — CHANGES_REQUESTED** [review]
Please clarify why those changes are necessary. For most of the changes, I couldn't see why they're justified, just by looking at the documentation for the original library `webpack-sources`.

File: types/webpack-sources/lib/CompatSource.d.ts (hunk 1 above)
Comment (SyMind) [reply types/webpack-sources/lib/CompatSource.d.ts:6]: `CompatSource` is inherits from `Source` so just return `Source` is ok.

Comment (SyMind) [reply types/webpack-sources/lib/CompatSource.d.ts:6]: @gabritto Another reason is that `webpack-sources` are designed for use in webpack, and the types in webpack are defined as such.

**Comment by SyMind** [comment]
@e-cloud @chriseppstein @peterblazejewicz cc

**Pull request closed without merging**